packages/dashboard/src/vaadin-dashboard-widget.js

```diff
@@ -259,14 +263,24 @@ class DashboardWidget extends DashboardItemMixin(ElementMixin(ThemableMixin(Poly
   }
 
   /** @private */
-  __onWidgetTitleChanged() {
-    this.__updateTitle();
+  __renderWidgetTitle() {
+    let effectiveHeadingLevel = this.__rootHeadingLevel;
+    // Default to 2 if not defined
+    if (effectiveHeadingLevel == null) {
+      effectiveHeadingLevel = 2;
+    }
+    // Increase level by 1 for widgets in sections
+    if (this.__isNestedWidget) {
+      effectiveHeadingLevel += 1;
+    }
+    return html`<div id="title" part="title" aria-level=${effectiveHeadingLevel} .hidden=${!this.widgetTitle}
+      >${this.widgetTitle}</div
+    >`;
```

Comment:
You can still keep using native heading elements here with

```js
const heading = document.createElement(`h${effectiveHeadingLevel}`);
heading.part = 'title';
heading.id = 'title';
heading.hidden = !this.widgetTitle;
heading.textContent = this.widgetTitle;
return heading;
```

If it's preferred to switch to using a `div` then ignore this

Reply:
It seems that there are cases where developers use higher `aria-level`s than 6 for their applications. In order to not limit the use, I switched to `div`. Do you think there would be any drawbacks to this?

Reply:
Probably not, just verify that the a11y tree and supported screen readers work as before.

Reply:
Added the heading role and the checked both. They seem to work as before.